internal/elasticsearch/ingest/datastream.go

```diff
@@ -16,12 +16,25 @@ import (
 	"strings"
 	"time"
 
+	"gopkg.in/yaml.v3"
+
 	"github.com/elastic/elastic-package/internal/elasticsearch"
 	"github.com/elastic/elastic-package/internal/packages"
 )
 
 var ingestPipelineTag = regexp.MustCompile(`{{\s*IngestPipeline.+}}`)
 
+type Rule struct {
+	TargetDataset interface{}   `yaml:"target_dataset"`
+	If            interface{}   `yaml:"if"`
+	Namespace     []interface{} `yaml:"namespace"`
```

Comment:
According the the package-spec 2.9.0:

- [target_dataset](https://github.com/elastic/package-spec/blob/067a0e5656a1603d03ff7f7589c1d9517becca3d/spec/integration/data_stream/routing_rules.spec.yml#L11-L27) and [namespace](https://github.com/elastic/package-spec/blob/067a0e5656a1603d03ff7f7589c1d9517becca3d/spec/integration/data_stream/routing_rules.spec.yml#L34-L51) can both be `string` or an array of `string`.
- [if](https://github.com/elastic/package-spec/blob/067a0e5656a1603d03ff7f7589c1d9517becca3d/spec/integration/data_stream/routing_rules.spec.yml#L28-L33) can only be a `string`

Maybe we can unmarshall them using a dedicated struct like:

```golang
type StringOrArray []string

func (s *StringOrArray) UnmarshalYAML(value *yaml.Node) error {
	var multi []string
	err := value.Decode(&multi)
	if err != nil {
		var single string
		err := value.Decode(&single)
		if err != nil {
			return err
		}
		*s = []string{single}
	} else {
		*s = multi
	}
	return nil
}

type Rule struct {
	TargetDataset StringOrArray `yaml:"target_dataset"`
	Namespace     StringOrArray `yaml:"namespace"`
	If            string        `yaml:"if"`
}
```

Reply:
Yep thats why I added the TODOs later. I'm trying to figure out what's the use case for an array of namespaces and target_dataset.

Reply:
> Yep thats why I added the TODOs later. 

Oh, god it! The question marks deceived me 😇 


> I'm trying to figure out the use case for an array of namespaces and target_dataset.

They can be pretty handy. The processor uses the first dataset and namespace with an actual value in the array, like the patterns in the grok. 

For example, in the Kubernetes use case:

```yaml
# Route container logs entries to the correct dataset and namespace based on the pod labels
- source_dataset: kubernetes.container_logs
  rules:
    - target_dataset:
        - "{{kubernetes.labels.elastic_co/dataset}}"
        - "{{data_stream.dataset}}"
        - kubernetes.container_logs
      namespace:
        - "{{kubernetes.labels.elastic_co/namespace}}"
        - "{{data_stream.namespace}}"
        - default
      if: "ctx?.kubernetes?.labels != null"
```

For the namespace, it will use the `{{kubernetes.labels.elastic_co/namespace}}` field if it exists, or try the next one until the literal value `default`.

Reply:
done 😬 Thanks!!!